### PR TITLE
fix OCP-21686  OCP-26640 + OCP-26404 OCP-23866

### PR DESCRIPTION
**Pull request by yuwang-RH**
fix OCP-21686  OCP-26640 + OCP-26404 OCP-23866

### CODE REVIEW

File: lib/rules/web/ocm_console/cluster_detail_access_control_tab.xyaml [hunk 1]
```diff
@@ -47,18 +47,36 @@ check_user_groups_in_select:
...
-      xpath: //button[@class='pf-c-button pf-m-primary' and text()='Add']
-click_add_button_in_user_tab:
-  element:
-    <<: *add_button_in_user_tab
-    op: click
 check_add_button_in_user_tab_disabled:
   element:
     selector:
       xpath: //button[@class='pf-c-button pf-m-primary' and @disabled and text()='Add']
+check_elements_on_add_user_dialog:
```
Comment (xueli181114) [inline lib/rules/web/ocm_console/cluster_detail_access_control_tab.xyaml:54]: This is a dialog element definition. Could you please move the related code to dialogs_and_sidebar.xyaml?

File: lib/rules/web/ocm_console/cluster_detail_access_control_tab.xyaml [hunk 2]
```diff
@@ -137,4 +153,13 @@ check_danger_alert_box_in_user_tab:
   - selector:
       xpath: //div[@class='pf-c-alert pf-m-inline pf-m-danger error-box' and @aria-label='Danger Alert']/div[@class='pf-c-alert__description']/span[contains(.,'<error_massage>')]
   - selector:
-      xpath: //div[@class='pf-c-alert pf-m-inline pf-m-danger error-box' and @aria-label='Danger Alert']/div[@class='pf-c-alert__description']/span[contains(.,'Operation ID')]
\ No newline at end of file
+      xpath: //div[@class='pf-c-alert pf-m-inline pf-m-danger error-box' and @aria-label='Danger Alert']/div[@class='pf-c-alert__description']/span[contains(.,'Operation ID')]
+click_add_user_button:
```
Comment (xueli181114) [inline lib/rules/web/ocm_console/cluster_detail_access_control_tab.xyaml]: better to name it to "click_add_user_button_on_page"

File: lib/rules/web/ocm_console/cluster_list_page.xyaml [hunk 3]
```diff
@@ -414,6 +414,7 @@ logout_page_loaded:
 go_to_osd_cluster_creation_page:
```
Comment (xueli181114) [inline lib/rules/web/ocm_console/cluster_list_page.xyaml]: We have a function switch_to_osd_creation_page.

File: lib/rules/web/ocm_console/creation_cards_page.xyaml [hunk 4]
```diff
@@ -33,6 +33,11 @@ click_create_osd_button:
         selector:
             xpath: //a[@class='infra-card pf-c-card create-cluster-card' and contains(@href, '/openshift/create/osd')]
         op: click
+click_create_osd_on_aws:
```
Comment (xueli181114) [inline lib/rules/web/ocm_console/creation_cards_page.xyaml]: duplicate with select_cloud_provider

**Review by xueli181114 — CHANGES_REQUESTED** [review]

File: lib/rules/web/ocm_console/cluster_detail_access_control_tab.xyaml (hunk 2 above)
Comment (yuwang-RH) [reply lib/rules/web/ocm_console/cluster_detail_access_control_tab.xyaml]: updated,thanks

File: lib/rules/web/ocm_console/cluster_list_page.xyaml (hunk 3 above)
Comment (yuwang-RH) [reply lib/rules/web/ocm_console/cluster_list_page.xyaml]: updated,thanks.

File: lib/rules/web/ocm_console/creation_cards_page.xyaml (hunk 4 above)
Comment (yuwang-RH) [reply lib/rules/web/ocm_console/creation_cards_page.xyaml]: updated

File: lib/rules/web/ocm_console/cluster_detail_access_control_tab.xyaml (hunk 1 above)
Comment (yuwang-RH) [reply lib/rules/web/ocm_console/cluster_detail_access_control_tab.xyaml:54]: The referenced elements are used in other places in the same file, so we need keep them in the same file here

**Pull request merged**